File: Groupe1/TP1/src/boucles.py
# Author: Matéo Guenot
# Date: 16/09/2024


def main():
    compteur = int(input("Entrer la largeur du triangle rectangle : "))

    if compteur < 1:
        print("Le nombre doit être supérieur à 0.")
        return

    if compteur >= 10:
        print("Le nombre doit être inférieur à 10.")
        return


    for i in range(1, compteur + 1):
        for j in range(1, i + 1):
            if j == 1 or j == i or i == compteur:
                print("*", end="")
            else:
                print("#", end="")
        print()

    i = 1
    while i <= compteur:
        j = 1
        while j <= i:
            if j == 1 or j == i or i == compteur:
                print("*", end="")
            else:
                print("#", end="")
            j += 1
        print()
        i += 1

# Appel de la fonction main
if __name__ == "__main__":
    main()

File: Groupe1/TP1/src/boucles.c
/**
 * Author:    Matéo Guenot
 * Created:   16/09/2024
 *
 **/

#include <stdio.h>

int main()
{
    // Déclaration et assignation des variables
    printf("Entrer la largeur du triangle rectangle : ");
    float largeur;
    scanf("%f", &largeur);

    if (largeur < 0)
    {
        printf("La largeur du triangle rectangle est négative\n");
        return 1;
    }

    if (largeur >= 10)
    {
        printf("La largeur du triangle rectangle est trop grande\n");
        return 1;
    }

    for (int i = 0; i <= largeur; i++)
    {
        for (int j = 0; j < i; j++)
        {
            if (j == 0 || j == i - 1 || i == largeur)
                printf("*");
            else
                printf("#");
        }
        printf("\n");
    }

    int i = 0;
    while (i <= largeur)
    {
        int j = 0;
        while (j < i)
        {
            if (j == 0 || j == i - 1 || i == largeur)
                printf("*");
            else
                printf("#");
            j++;
        }
        printf("\n");
        i++;
    }

    return 0;
}